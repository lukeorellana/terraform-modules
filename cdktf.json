{
  "language": "typescript",
  "app": "npx ts-node",
  "terraformProviders": [],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb"
}