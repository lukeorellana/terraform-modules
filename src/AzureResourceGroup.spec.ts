import { Testing, TerraformStack} from 'cdktf';
import {AzurermProvider} from "@cdktf/provider-azurerm";
import { AzureResourceGroup } from './AzureResourceGroup';
import 'cdktf/lib/testing/adapters/jest';


// test('renders a VPC with minimal config', () => {
//   const synthed = Testing.synthScope((stack) => {
//     new PocketVPC(stack, 'testPocketVPC');
//   });


describe('AzureResourceGroup-Snapshot', () => {
  it('renders a AzureResourceGroup without tags', () => {

    const synthed = Testing.synthScope((stack) => {
      new AzureResourceGroup(stack, 'testRG', {
        name: 'rg-test',
        location: 'eastus',
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders a AzureResourceGroup with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new AzureResourceGroup(stack, 'testRG', {
        name: 'rg-test',
        location: 'eastus',
        tags: {
          name: 'rug',
          description: 'test environmnet',
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureResourceGroup-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });


    new AzureResourceGroup(stack, 'testRG', {
      name: 'rg-test',
      location: 'eastus',
      tags: {
        name: 'test',
        Env: "NonProd",
      },
    });


    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });

    new AzureResourceGroup(stack, 'testRG', {
      name: 'rg-test',
      location: 'eastus',
      tags: {
        name: 'test',
        Env: "NonProd",
      },
    });



    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
