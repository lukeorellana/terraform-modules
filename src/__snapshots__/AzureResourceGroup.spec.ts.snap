// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureResourceGroup-Snapshot renders a AzureResourceGroup with tags 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.id}"
    },
    "location": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.location}"
    },
    "name": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.name}"
    }
  },
  "resource": {
    "azurerm_resource_group": {
      "testRG_rg_D9B234FB": {
        "lifecycle": [
          {
            "ignore_changes": [
            ]
          }
        ],
        "location": "eastus",
        "name": "rg-test",
        "tags": {
          "description": "test environmnet",
          "name": "rug"
        }
      }
    }
  }
}"
`;

exports[`AzureResourceGroup-Snapshot renders a AzureResourceGroup without tags 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.id}"
    },
    "location": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.location}"
    },
    "name": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.name}"
    }
  },
  "resource": {
    "azurerm_resource_group": {
      "testRG_rg_D9B234FB": {
        "lifecycle": [
          {
            "ignore_changes": [
            ]
          }
        ],
        "location": "eastus",
        "name": "rg-test"
      }
    }
  }
}"
`;
